src/win/poll.c

```diff
@@ -583,34 +484,34 @@ int uv_poll_init_socket(uv_loop_t* loop, uv_poll_t* handle,
 
 
 int uv_poll_start(uv_poll_t* handle, int events, uv_poll_cb cb) {
-  int err;
+  assert(handle->type == UV_POLL);
+  assert(!(handle->flags & UV_HANDLE_CLOSING));
+  assert((events & ~(UV_READABLE | UV_WRITABLE | UV_DISCONNECT)) == 0);
 
-  if (!(handle->flags & UV_HANDLE_POLL_SLOW)) {
-    err = uv__fast_poll_set(handle->loop, handle, events);
+  handle->events = events;
+  handle->poll_cb = cb;
+
+  if (handle->events != 0) {
+    uv__handle_start(handle);
   } else {
-    err = uv__slow_poll_set(handle->loop, handle, events);
+    uv__handle_stop(handle);
   }
 
-  if (err) {
-    return uv_translate_sys_error(err);
+  if ((handle->events & ~(handle->submitted_events_1 |
+      handle->submitted_events_2)) != 0) {
+    if (handle->flags & UV_HANDLE_POLL_SLOW) {
+      uv__slow_poll_submit_poll_req(handle->loop, handle);
+    } else {
+      uv__fast_poll_submit_poll_req(handle->loop, handle);
+    }
   }
 
-  handle->poll_cb = cb;
-
   return 0;
 }
 
 
 int uv_poll_stop(uv_poll_t* handle) {
-  int err;
-
-  if (!(handle->flags & UV_HANDLE_POLL_SLOW)) {
-    err = uv__fast_poll_set(handle->loop, handle, 0);
-  } else {
-    err = uv__slow_poll_set(handle->loop, handle, 0);
-  }
-
-  return uv_translate_sys_error(err);
+  return uv_poll_start(handle, 0, handle->poll_cb);
```

Comment:
Calling `uv_poll_start()` in `uv_poll_stop()` seems odd to me. I think it's better to make the contents of `uv_poll_start()` into `uv__poll_set()` and call it from `uv_poll_start()` and `uv_poll_stop()`.

Reply:
Updated *shrug*